tests/src/Integration/Action/EntityFetchByFieldTest.php

```diff
@@ -0,0 +1,209 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\Tests\rules\Integration\Action\EntityPropertyFetchTest.
+ */
+
+namespace Drupal\Tests\rules\Integration\Action;
+
+use Drupal\Tests\rules\Integration\RulesEntityIntegrationTestBase;
+
+/**
+ * @coversDefaultClass \Drupal\rules\Plugin\Action\EntityFetchByField
+ * @group rules_action
+ */
+class EntityPropertyFetchTest extends RulesEntityIntegrationTestBase {
+
+  /**
+   * The action to be tested.
+   *
+   * @var \Drupal\rules\Engine\RulesActionInterface
+   */
+  protected $action;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setUp() {
+    parent::setUp();
+
+    // Prepare our own dummy entityManager as the entityManager in
+    // RulesEntityIntegrationTestBase does not mock the getStorage method.
+    $this->entityManager = $this->getMockBuilder('Drupal\Core\Entity\EntityManager')
+      ->setMethods(['getBundleInfo', 'getStorage'])
+      ->setConstructorArgs([
+        $this->namespaces,
+        $this->moduleHandler,
+        $this->cacheBackend,
+        $this->getMock('Drupal\Core\Language\LanguageManagerInterface'),
+        $this->getStringTranslationStub(),
+        $this->getClassResolverStub(),
+        $this->typedDataManager,
+        $this->getMock('Drupal\Core\KeyValueStore\KeyValueStoreInterface'),
+        $this->getMock('Symfony\Component\EventDispatcher\EventDispatcherInterface')
+      ])
+      ->getMock();
+
+    // Return some dummy bundle information for now, so that the entity manager
+    // does not call out to the config entity system to get bundle information.
+    $this->entityManager->expects($this->any())
+      ->method('getBundleInfo')
+      ->with($this->anything())
+      ->willReturn(['entity_test' => ['label' => 'Entity Test']]);
+    $this->container->set('entity.manager', $this->entityManager);
+
+    $this->action = $this->actionManager->createInstance('rules_entity_property_fetch');
+  }
+
+  /**
+   * Tests the summary.
+   *
+   * @covers ::summary
+   */
+  public function testSummary() {
+    $this->assertEquals('Fetch entities by property', $this->action->summary());
+  }
+
+  /**
+   * Tests action execution when no value for limit is provided.
+   *
+   * @covers ::execute
+   */
+  public function testActionExecutionWithNoLimit() {
+    // Create variables for action context values.
+    $entity_type = 'entity_test';
+    $property_name = 'test_property';
```

Comment:
I've changed them to `field_name` and `field_value` (rather than `property` and `value`). I think this is clearer.